sdks/java/core/src/main/java/org/apache/beam/sdk/Pipeline.java

```diff
@@ -380,6 +524,32 @@ public void traverseTopologically(PipelineVisitor visitor) {
   }
 
   /**
+   * retrieve {@code Pipeline}, and print out data flow in visualized format.
+   *
+   * <p>Give a pipeline like
+   * <pre>{@code
+   * Pipeline pipeline = Pipeline.create(...);
+   * PCollection<?> stream = pipeline.apply("source",IO.<?>read())
+   *   .apply("process_step", ParDo.of(...));
+   * stream.apply("persistent", IO.<?>write());
```

Comment:
What does this look like if there's branches or joins? Especially if something ends with high fanout, or a large degree of independent paths?